src/hist.jl

```diff
@@ -14,6 +16,31 @@ function _check_closed_arg(closed::Symbol, funcsym)
 end
 
 
+## Fast map for tuples (a lot faster than using Base.map)
+@generated function _tuple_map{N}(f, tuples::NTuple{N, Any}...)
```

Comment:
I took it out - surprisingly, performance in the benchmark given above is still about the same (but in my tuple-map test, there is a vast difference in performance). While I developed this, I did need _tuple_map to make histogram filling fast - but it seems like in the final implementation now, it doesn't make much of a difference. Maybe inlining behaviour somewhere - don't know. Anyhow, looks good. (I'll play a bit more with tuple mapping, independent of StatsBase, and think about a PR for Base.)